mesa/agent.py

```diff
@@ -304,6 +304,25 @@ def map(self, method: str | Callable, *args, **kwargs) -> list[Any]:
 
         return res
 
+    def agg(self, attribute: str, func: Callable) -> Any:
+        """
+        Aggregate an attribute of all agents in the AgentSet using a specified function.
+
+        Args:
+            attribute (str): The name of the attribute to aggregate.
+            func (Callable): The function to apply to the attribute values (e.g., min, max, sum, np.mean).
+
+        Returns:
+            Any: The result of applying the function to the attribute values.
+        """
+        # Retrieve the attribute values from all agents
+        values = [
+            getattr(agent, attribute) for agent in self if hasattr(agent, attribute)
```

Comment:
You are passing silently over the agents that don't have the attribute. Not sure about this. It seems at odds with the zen of python.

Reply:
Good point. That's indeed way more elegant. It also raises an `AttributeError` if an Agent doesn't have the value.